discover-cpu-partitioning.py

```diff
@@ -0,0 +1,173 @@
+        print("online cpus: %d" % (len(online_cpus)))
+        print("")
+        rcu_nocbs_cpus = get_rcu_nocbs()
+        print("rcu_nocbs: %s" % (rcu_nocbs_cpus))
+        print("rcu_nocbs: %s" % (system_cpu_topology.formatted_cpu_list(rcu_nocbs_cpus)))
+        print("rcu_nocbs: %d" % (len(rcu_nocbs_cpus)))
+        print("")
+
+        odd_list = list(set(rcu_nocbs_cpus) - set(online_cpus))
+        if len(odd_list) > 0:
+            print("this is odd, rcu_nocbs contains cpus that are not online: %s" % (odd_list))
+
+        housekeeping_cpus = list(set(online_cpus) - set(rcu_nocbs_cpus))
+        if len(housekeeping_cpus) == 0:
+            print("this is odd, there are no housekeeping cpus")
+        else:
+            print("housekeeping cpus: %s" % (housekeeping_cpus))
+            print("housekeeping cpus: %s" % (system_cpu_topology.formatted_cpu_list(housekeeping_cpus)))
+            print("housekeeping cpus: %d" % (len(housekeeping_cpus)))
+            print("")
+            print("isolated cpus:     %s" % (rcu_nocbs_cpus))
+            print("isolated cpus:     %s" % (system_cpu_topology.formatted_cpu_list(rcu_nocbs_cpus)))
+            print("isolated cpus:     %d" % (len(rcu_nocbs_cpus)))
+    elif t_global.args.environment == "container":
+        system_cpus = system_cpu_topology(debug = t_global.args.debug)
+
+        all_cpus = system_cpus.get_all_cpus()
+        print("all cpus: %s" % (all_cpus))
+        print("all cpus: %s" % (system_cpu_topology.formatted_cpu_list(all_cpus)))
+        print("all cpus: %d" % (len(all_cpus)))
+        print("")
+        online_cpus = system_cpus.get_online_cpus()
+        print("online cpus: %s" % (online_cpus))
+        print("online cpus: %s" % (system_cpu_topology.formatted_cpu_list(online_cpus)))
+        print("online cpus: %d" % (len(online_cpus)))
+        print("")
+        
+        try:
+            cpus_allowed = get_pid_cpus_allowed('self')
+        except Exception as e:
+            print(e)
+            return(1)
+
+        housekeeping_cpus = []
+        isolated_cpus = []
+        if len(cpus_allowed) == 1:
+            print("this is odd, there really needs to be more than 1 CPU in the cpus_allowed list for proper functionality")
+            housekeeping_cpus = copy.deepcopy(cpus_allowed)
+            isolated_cpus = copy.deepcopy(cpus_allowed)
+        else:
+            housekeeping_cpus.append(cpus_allowed.pop(0))
+            isolated_cpus = copy.deepcopy(cpus_allowed)
+
+        print("housekeeping cpus: %s" % (housekeeping_cpus))
+        print("housekeeping cpus: %s" % (system_cpu_topology.formatted_cpu_list(housekeeping_cpus)))
+        print("housekeeping cpus: %d" % (len(housekeeping_cpus)))
+        print("")
+        print("isolated cpus:     %s" % (isolated_cpus))
+        print("isolated cpus:     %s" % (system_cpu_topology.formatted_cpu_list(isolated_cpus)))
+        print("isolated cpus:     %d" % (len(isolated_cpus)))
```

Comment:
Is this going to set env vars, or is bootstrap going to check output and then set env vars?

Reply:
bootstrap is going to check the output and set the vars.  bootstrap has to do it so that children of bootstrap can see and inherit the values.  If this script did it then it would disappear as soon as this script exited.

Reply:
OK, sounds good.  I assume bootstrap will also taskset all pids in the container (if this is a container) or any child pids (if on baremetal) to the hk_cpus before running client-server-script.

Reply:
bootstrap tasksets itself so that anything it forks will inherit the new cpu mask:

https://github.com/perftool-incubator/rickshaw/blob/6facf2fbcdb4155119c8e09514a4428c46f53267/bootstrap#L193